apps/directory/src/stores/biobanksStore.js

```diff
@@ -117,7 +117,14 @@ export const useBiobanksStore = defineStore("biobanksStore", () => {
 
       /* Update biobankCards only if the result is the most recent one*/
       if (requestTime === lastRequestTime) {
-        biobankCards.value = filterWithdrawn(biobankResult.Biobanks || []);
+        let biobanksWithCollections = biobankResult.Biobanks;
+        if (!biobanksWithCollections) biobanksWithCollections = [];
+        if (filtersStore.hasActiveFilters) {
+          biobanksWithCollections = biobanksWithCollections.filter(
+            (biobank) => biobank.collections?.length
+          );
+        }
+        biobankCards.value = filterWithdrawn(biobanksWithCollections);
```

Comment:
Please use variable/function names to describe what the code does. Don't keep overwriting the same variable over and over.